src/main/java/ListOfItems.java

```diff
@@ -0,0 +1,353 @@
+        System.out.println(divider);
+        if (list.size() == 0) {
+            throw new DukeException("List is empty, you have free time (for now)! YAY :D" + "\n" + divider);
+        } else {
+            System.out.println("Here are the task(s) in your list: ");
+            for (int i = 0; i < index; i++) {
+                System.out.println(list.get(i).id + "." + list.get(i));
+            }
+        }
+        System.out.println(divider);
+    }
+
+    /**
+     * Marks a particular task is done.
+     *
+     * @param input user input.
+     * @throws DukeException if number given is invalid.
+     */
+    protected void doneItem(String input) throws DukeException {
+        try {
+            // retrieve number after "done "
+            int number = Integer.parseInt(input.substring(5));
+
+            Task task = list.get(number - 1);
+            if (task.isDone) {
+                System.out.println("Task already done!");
+            } else {
+                task.markedDone();
+                String message = "Good job! I've marked this task as done: ";
+                System.out.println(divider + "\n" + message + "\n" + tabSpacing
+                        + task + "\n" + divider);
+            }
+        } catch (IndexOutOfBoundsException | NumberFormatException e) {
+            throw new DukeException("\n" + divider + "\n" + "Sorry, you did not enter a valid number. Please try again."
+                    + "\n" + divider);
+        }
+    }
+
+    /**
+     * Deletes a particular task.
+     * Modifies other task's index if necessary, so that list is still in chronological order.
+     *
+     * @param input user input.
+     * @throws DukeException if number given is invalid.
+     */
+    protected void deleteItem(String input) throws DukeException {
+        try {
+            int number = Integer.parseInt(input.substring(7));
+            Task task = list.get(number - 1);
+            for (int i = number; i < list.size(); i++) {
+                list.get(i).id = list.get(i).id - 1;
+            }
+            list.remove(task);
+            index--;
+            System.out.println(divider);
+            System.out.println("Noted. I've removed this task: ");
+            System.out.println(tabSpacing + task);
+            System.out.println("Now you have " + index + " tasks in the list.");
+            System.out.println(divider);
+        } catch (IndexOutOfBoundsException | NumberFormatException e) {
+            throw new DukeException("\n" + divider + "\n" + "Whoops, you did not enter a valid number."
+                    + "\n" + divider);
+        }
+    }
+
+    /**
+     * Adds a new task to the list.
+     * Checks what type of task is given and initialise a new sub-class of Task (To-do, Deadline, or Event).
+     *
+     * @param input user input.
+     * @throws DukeException if incomplete commands are given.
+     */
+    protected void addItem(String input) throws DukeException {
```

Comment:
This method seems to be doing too much work, perhaps can consider abstracting the parsing into the Parser class? I noticed this issue in several other places.